hotdoc/extensions/gi/gi_extension.py

```diff
@@ -233,7 +234,10 @@ def _get_smart_key(self, symbol):
                                 super()._get_smart_key(symbol))
 
     def _get_all_sources(self):
-        return [s for s in self.c_sources if s.endswith('.h')]
+        if not self.__all_sources:
+            self.__all_sources = list({s.replace('.c', '.h') for s in self.c_sources})
```

Comment:
That's very weird, did we end up finding another way to do that? If not, we probably should discuss that :)

Reply:
Fixed the way we agreed on (c files keep being referenced as `.c`).